src/store/arweave-report-sink.ts

```diff
@@ -0,0 +1,161 @@
+/**
+ * AR.IO Observer
+ * Copyright (C) 2023 Permanent Data Solutions, Inc. All Rights Reserved.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+import Arweave from 'arweave';
+import { promisify } from 'node:util';
+import zlib from 'node:zlib';
+import * as winston from 'winston';
+
+import { REPORT_FORMAT_VERSION } from '../observer.js';
+import { ObserverReport, ReportInfo, ReportSink } from '../types.js';
+
+const gzip = promisify(zlib.gzip);
+
+export class ArweaveReportSink implements ReportSink {
+  // Dependencies
+  private log: winston.Logger;
+  private arweave: Arweave;
+  private readonly walletJwk: any; // JWK wallet type
+
+  constructor({
+    log,
+    arweave,
+    walletJwk,
```

Comment:
for other wallet support we may want to add providing a signer here - and it will sign the data item before posting to the arweave client, but otherwise looks good!

Reply:
Yeah, won't worry about that now, but may get something in for it after TGE.